dragonfly/implementations/PsyllidProvider.py

```diff
@@ -0,0 +1,404 @@
+'''
+'''
+
+from __future__ import absolute_import
+
+# standard imports
+import logging
+import uuid
+import time
+import os
+import json
+from datetime import datetime
+
+# internal imports
+from dripline import core
+from dragonfly.implementations import EthernetProvider
+
+
+__all__ = []
+
+logger = logging.getLogger(__name__)
+
+__all__.append('PsyllidProvider')
+class PsyllidProvider(core.Provider, core.Spime):
```

Comment:
If you don't know a reason then there probably is none. When I wrote that I looked at what some of the other classes inherited from at the time.